packages/babel-plugin-transform-react-constant-elements/src/index.js

```diff
@@ -53,6 +55,19 @@ export default function ({ types: t }) {
         if (path.node._hoisted) return;
 
         const state = { isImmutable: true };
+
+        // This transform takes the option `allowMutablePropsOnTags`, which is an array
+        // of JSX tags to allow mutable props (such as objects, functions) on. Use sparingly
+        // and only on tags you know will never modify their own props.
+        if (this.opts.allowMutablePropsOnTags != null) {
+          if (!Array.isArray(this.opts.allowMutablePropsOnTags)) {
+            throw new Error(".allowMutablePropsOnTags must be an array, null, or undefined.");
+          }
+          const elementName = path.node.openingElement.name.name;
```

Comment:
Good question. I guess I was thinking like `Intl` or `Intl.*` as saying a whole namespace, with `Intl.FormattedMessage` being required for the specific tag. Guess it might be nice to allow `FormattedMessage` if people are destructuring out of `Intl` though.

What do you think is best?